### ISSUE TITLE
For in loop type inference only works for arrays

### ISSUE THREAD

**Issue by jaypavlina** [HaxeFoundation/intellij-haxe#25]
Type inference works fine for the following because it's an array. Item will be recognized as a String and will give code completion for the "indexOf" method.

```
var array : Array<String> = ["item"];
for (item in array) {
    var index = item.indexOf("e"); // works fine
}
```

However, on a map, the same thing does not work. The type of "key" is unknown, so there is no code completion, and calling "indexOf" gives an unresolved symbol warning.

```
var map : Map<String,String> = ["key" => "value"];
for (key in map.keys()) {
    var index = key.indexOf("e"); // unresolved symbol warning
}
```

I am not sure if it works on other types of iterators, but it doesn't work on Map.keys() or Map.iterator(), so I assume it's not working for any others.

TIR: STB-7289

**Comment by EBatTiVo** [comment]
At a guess, I think this may be due to a lack of -- or broken -- generics (type parameter) support.

**Comment by EricBishton** [comment]
Took a while, but it's fixed in 0.11.0-RC3.  https://github.com/HaxeFoundation/intellij-haxe/commit/f128e86bf0fa88a8b6d8dfd66e2fa63531c9a18a
